client.go

```diff
@@ -38,6 +39,10 @@ var (
 	// defaultClient is used for performing requests without explicitly making
 	// a new client. It is purposely private to avoid modifications.
 	defaultClient = NewClient()
+
+	// We need to consume response bodied to maintain http connections, but
```

Comment:
Misspelled "bodies" (?). I also wonder if it makes sense to limit this, as it makes it a bit unpredictable what the client will actually do or when it will re-use a connection. Thoughts?